os/src/ZipOps.scala

```diff
@@ -153,17 +154,19 @@ object zip {
     val mtimeOpt = if (preserveMtimes) Some(os.mtime(file)) else None
 
     val fis = if (os.isFile(file)) Some(os.read.inputStream(file)) else None
-    try makeZipEntry0(sub, fis, mtimeOpt, zipOut)
+
+    val path = if (os.isDir(file)) sub.toString + "/" else sub.toString
+    try makeZipEntry0(path, fis, mtimeOpt, zipOut)
     finally fis.foreach(_.close())
   }
 
   private def makeZipEntry0(
-      sub: os.SubPath,
+      path: String,
```

Comment:
Why are we using `String` here?

Reply:
Because treats a `ZipEntry` as directory if its name ends with "/" characters.
Firstly I want to add end "/" character, but in `os.Path` it's not valid.

Reply:
I know it's a good way to implement, since it breaks the type check inside `os.Path`, could you give me some advice ?
should I rewrite another method to handle folders such like `makeZipFolderEntry0` ?

Reply:
We should be able to keep the conversion to string and appending of `/` inside `makeZipEntry0`. `is: Option` should already tell us whether it's a file or directory I think

Reply:
Done. But I'm afraid This might be a little slightly fragile assumption, since it's relied on
```scala
val fis = if (os.isFile(file)) Some(os.read.inputStream(file)) else None
```